library/core/src/ptr/non_null.rs

```diff
@@ -502,6 +502,11 @@ impl<T: ?Sized> NonNull<T> {
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
     #[stable(feature = "non_null_convenience", since = "1.80.0")]
     #[rustc_const_stable(feature = "non_null_convenience", since = "1.80.0")]
+    #[requires(!origin.as_ptr().is_null())]
```

Comment:
`!self.as_ptr().is_null()` is not necessary because NonNull guarentees no null pointer. Please consider other properties.

Reply:
After discuss with sponsor on meeting, we should keep it.